lisa/sut_orchestrator/qemu/platform.py

```diff
@@ -462,11 +520,26 @@ def _create_node_cloud_init_iso(
         user_data_string = "#cloud-config\n" + yaml.safe_dump(user_data)
         meta_data_string = yaml.safe_dump(meta_data)
 
+        iso_path = node_context.cloud_init_file_path
+        tmp_dir = None
+        if self.qemu_platform_runbook.is_host_remote():
+            tmp_dir = tempfile.TemporaryDirectory()
+            iso_path = os.path.join(tmp_dir.name, "cloud-init.iso")
+
         self._create_iso(
-            node_context.cloud_init_file_path,
+            iso_path,
             [("/user-data", user_data_string), ("/meta-data", meta_data_string)],
         )
 
+        if self.qemu_platform_runbook.is_host_remote():
+            host = self.qemu_platform_runbook.host
+            local_node.execute(
+                f"scp -i {host.private_key_file} "
+                f"{iso_path} "
+                f"scp://{host.address}/{node_context.cloud_init_file_path}"
+            )
+            tmp_dir.cleanup()
```

Comment:
The `tmp_dir.cleanup()` should be run in a `finally` block, to ensure it is always removed even if there are unexpected exceptions.

For example,
```python3
try:
    tmp_dir = None
    if self.qemu_platform_runbook.is_host_remote():
        tmp_dir = tempfile.TemporaryDirectory()
        ...
finally:
    if tmp_dir:
        tmp_dir.cleanup()

```